src/pyannote/audio/__main__.py

```diff
@@ -612,54 +618,73 @@ def benchmark(
     # manual annotation and j speakers in the prediction
     speaker_count: dict[int, dict[int, int]] = dict()
 
-    with open(into / f"{benchmark_name}.rttm", "w") as rttm:
-        # iterate over all files in the specified subset
-        for file in track(files, disable=not progress):
-            # gather file metadata
-            uri: str = file["uri"]
-            playing_time[uri] = Audio().get_duration(file)
-
-            tic: float = time.time()
-
-            # apply pretrained pipeline to file
-            prediction = pretrained_pipeline(file, **file.get("pipeline_kwargs", {}))
-
-            tac: float = time.time()
-            processing_time[uri] = tac - tic
-
-            # if prediction has a built-in serialize method, save serialized version
-            if hasattr(prediction, "serialize"):
+    if per_file:
+        rttm_dir = into / benchmark_name / "rttm"
+        rttm_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
Same. We should probably raise if `into / benchmark_name` exists (for consistency with the other case).